OneCommunityGlobal/HighestGoodNetworkApp
# Navneeth create see teams management tab custom permission

# Description
![feature description](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/59dad3e1-0f18-4604-ad43-51bcf2871596)



## Related PRs:
This PR is linked to the BE PR [#556](https://github.com/OneCommunityGlobal/HGNRest/pull/556).


## Main changes explained:
- Employing consistent naming of permissions in `src/utils/routePermissions.js`.
- Deciding that when a volunteer account has the `See Teams Management Tab (Create Teams and edit team members only)`, then the account will be able to add teams and add/delete members. 
- Rendering the UI according to the permissions wherein an account with just the `See Teams Management Tab (Create Teams and edit team members only)` permission will not be able to delete teams and edit team names.
- Updating the permission description for the `See Teams Management Tab (Create Teams and edit team members only)` permission.

## How to test:
1. Follow the steps in the backend (BE PR [#556](https://github.com/OneCommunityGlobal/HGNRest/pull/556)) to run the relevant PR.
2. In the frontend, check into the current branch.
3. do `npm install` and `npm run start:local` to run this PR locally
4. Log in as Owner -> Dashboard -> Other Links -> Permissions Management -> Manage User Permissions
5. Enter the username of an account that does not have the `See Teams Management Tab (Create Teams and edit team members only)` permission. 
6. Next to `See Teams Management Tab (Create Teams and edit team members only)`, click on Add. Then scroll down and click on Submit.
7. Log in as the volunteer account (if already logged in, make sure to log out and log back in). The `Other Links` button should appear with ONLY the `Teams` option visible/accessible.
8. Verify that the volunteer account can add teams and add/delete members in the Teams Management page (Teams).

## Test Video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/460d9abf-895e-4ed5-a85d-f2a8eedc98b2
